src/components/Icon/Icon.js

```diff
@@ -0,0 +1,69 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import bem from 'bem';
+import styles from './Icon.scss';
+import { CONTEXTS, ICONS } from '../../constants';
+
+const { block, elem } = bem({
+    name: 'Icon',
+    classnames: styles,
+    propsToMods: ['context', 'margin']
+});
+
+const Icon = props => {
+    const { context, name, size, title, ...rest } = props;
+
+    const currentIcon = ICONS[name];
+
+    return (
+        <div {...rest} {...block(props)}>
+            <svg
+                {...elem('svg', props)}
+                xmlns="http://www.w3.org/2000/svg"
+                viewBox={currentIcon.viewBox}
+                aria-labelledby="title"
+                style={(() => {
```

Comment:
Inline functions inside a render method triggers unnecessaery rerenderings as far as they being recreatred on each render. Better to extract it to something like `function calculateSizes(size) { ... }`.

And also it makes code herder to read.

Reply:
Would you split it off into a separate util or have it as a class method?

Reply:
I'd just declare a function before in the same file
```
const Icon = props => {
```